### ISSUE TITLE
Change message retrieval workflow

### ISSUE THREAD

**Issue by jsalhi** [jsalhi/ListenUp#12]
Messages are retrieved when their view in the adapter is called upon. This should change, as messages should be retrieved when the application is opened and whenever a new message is received

**Comment by yolkfolk** [comment]
Fixed :: c5d255f4f3e14c6eb6ebc5886ba510e9a90439be

We need to make this more efficient though. I've added notes as comments in the code.

**Comment by jsalhi** [comment]
NO. 

Messages should be retrieved:

-When the application is opened
-When a new message is received
-When some sort of manual update function/button is activated (Snapchat does this will the pulldown animation of the dancing ghost)

In addition, only NEW messages that are not already available on the local device should be retrieved

**Comment by yolkfolk** [comment]
I've addressed 2) in the comments of the code, with a suggestion on how we could fix it. We'll probably need to cache messages and maintain some sort of internal state on the device that tells it which messages have been loaded, since we want messages to be re-loaded if the user logs in with an entirely difference device. There are several components to the problem, which is why I opted for the most simplistic (and admittedly inefficient) approach available for now.

1) and 3) seem like separate issues.
